src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -53,17 +54,22 @@ class ArrayBuffer[A] private (initialElements: Array[AnyRef], initialSize: Int)
 
   /** Ensure that the internal array has at least `n` cells. */
   protected def ensureSize(n: Int): Unit =
-    array = RefArrayUtils.ensureSize(array, size0, n)
+    array = ArrayBuffer.ensureSize(array, size0, n)
 
   def sizeHint(size: Int): Unit =
     if(size > length && size >= 1) ensureSize(size)
 
   /** Reduce length to `n`, nulling out all dropped elements */
   private def reduceToSize(n: Int): Unit = {
-    RefArrayUtils.nullElems(array, n, size0)
+    Arrays.fill(array, n, size0, null)
     size0 = n
   }
 
+  /** Shrinks the ArrayBuffer to an appropriate size for the current number of elements (rounding up to the next
+    * natural size), which may replace the array by a shorter one. This allows releasing some unused memory. */
+  def shrinkToSize(): Unit =
```

Comment:
Java calls this `trimToSize`. Is there a reason to deviate from that? - https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#trimToSize--